packages/hash/api/src/model/knowledge/orgMembership.model.ts

```diff
@@ -0,0 +1,170 @@
+    graphApi: GraphApi,
+    params: OrgMembershipModelCreateParams,
+  ) {
+    const { responsibility, org } = params;
+
+    const properties: object = {
+      [WORKSPACE_TYPES.propertyType.responsibility.baseUri]: responsibility,
+    };
+
+    const entityTypeModel = WORKSPACE_TYPES.entityType.orgMembership;
+
+    const entity = await EntityModel.create(graphApi, {
+      accountId: workspaceAccountId,
+      properties,
+      entityTypeModel,
+    });
+
+    await entity.createOutgoingLink(graphApi, {
+      linkTypeModel: WORKSPACE_TYPES.linkType.ofOrg,
+      targetEntityModel: org,
+    });
+
+    return new OrgMembershipModel({
+      accountId: workspaceAccountId,
+      entityId: entity.entityId,
+      version: entity.version,
+      entityTypeModel,
+      properties,
+    });
+  }
+
+  /**
+   * Get a workspace organization entity by its entity id.
+   *
+   * @param params.entityId - the entity id of the organization
+   */
+  static async getOrgMembershipById(
+    graphApi: GraphApi,
+    params: { entityId: string },
+  ): Promise<OrgMembershipModel | null> {
+    const entity = await EntityModel.getLatest(graphApi, {
+      // assumption: `accountId` of organizations is always the workspace account id
+      accountId: workspaceAccountId,
+      entityId: params.entityId,
+    });
+
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.orgMembership.schema.$id
+    ) {
+      throw new Error(
+        `Entity with id ${params.entityId} is not a workspace org membership`,
+      );
+    }
+
+    return entity ? new OrgMembershipModel(entity) : null;
+  }
+
+  /**
+   * Get the org linked to the org membership.
+   */
+  async getOrg(graphApi: GraphApi): Promise<OrgModel> {
+    const { data: outgoingOrgLinks } = await graphApi.getLinksByQuery({
+      all: [
+        {
+          eq: [{ path: ["source", "id"] }, { literal: this.entityId }],
+        },
+        {
+          eq: [
+            { path: ["type", "versionedUri"] },
+            {
+              literal: WORKSPACE_TYPES.linkType.ofOrg.schema.$id,
+            },
+          ],
+        },
+      ],
+    });
+
+    const [outgoingOrgLink] = outgoingOrgLinks;
+
+    if (!outgoingOrgLink) {
+      /**
+       * This should never be the case, as the `org` link is required on `OrgMembership` entities.
+       *
+       * @todo: potentially remove this when the Graph API validates entities based on their schema
+       */
+      throw new Error(
+        `Critical: org membership with entity id ${this.entityId} doesn't have an outgoing "org" link`,
+      );
+    }
+
+    const { targetEntityId: orgEntityId } = outgoingOrgLink;
+
+    const orgModel = await OrgModel.getOrgById(graphApi, {
+      entityId: orgEntityId,
+    });
+
+    if (!orgModel) {
+      throw new Error(
+        `Critical: org membership with entity id ${this.entityId} links to non-existent org with id ${orgEntityId}`,
+      );
```

Comment:
How this can happen?

Reply:
It should never happen if the Graph API enforces constraints on link targets correctly. I instinctively threw an error to avoid casting here, would you like me to cast instead?

Reply:
A better idea may be to make use of the `Link` model class, by implementing a `link.getTarget()` method that returns the target entity. Then we can encapsulate the type casting (or the error in that method).

Let me know if that sounds like a good idea to you, in which case I'll implement it as part of this PR.

Reply:
I think that sounds like the way to go 👍🏻

Reply:
Have addressed in:
- https://github.com/hashintel/hash/pull/1068/commits/2ccb9ad47dc1272ac6acbcaa6589236cbcf59cb8
- https://github.com/hashintel/hash/pull/1068/commits/9a8d0016dbcc668e70191295c0088f8730e6b201